doc/development/deprecations.rst

```diff
@@ -63,6 +63,12 @@ Pending deprecations
   - Deprecated in v0.37
   - Will be removed in v0.39
 
+* The ``BasisStatePreparation`` template is deprecated.
+  Instead, ``BasisState`` can be called on the constructed operator.
+
+  - Deprecated in v0.38
+  - Will be removed in v0.39
```

Comment:
Double check

Reply:
Works for me! Any concerns @isaacdevlugt?

Reply:
Thanks @KetpuntoG! I think it's fine to deprecate this for v0.38, remove for v0.39. I created a story in the deps / removals for v0.38 that relates to the story Guillermo is working on :)